public/static/app/perfdiscovery/PerformanceDiscoveryCtrl.js

```diff
@@ -92,6 +92,39 @@ mciModule.controller('PerformanceDiscoveryCtrl', function (
     vm.updateData(true);
   }
 
+  $scope.changeMetric = function () {
+    const fromVersion = vm.fromSelect.selected
+    const toVersion = vm.toSelect.selected
+    const expandedCurrent = vm.expandedOptions && vm.expandedOptions.includes("current");
+    const expandedBaseline = vm.expandedOptions && vm.expandedOptions.includes("baseline");
+    const expandedHistory = vm.expandedOptions && vm.expandedOptions.includes("history");
+    updateColumnDefs(vm.gridApi.grid);
+
+    vm.isLoading = true;
+    $q.all({
+        fromVersionObj: dataUtil.getCompItemVersion(fromVersion),
```

Comment:
Are the __dataUtil.getCompItemVersion__ calls needed? loadCompOptions use the returned promise, but this version does not AFAICS.

I think you could replace with the following:

```
      dataUtil.changeMetric(vm.metric_name)
      .then(function (res) {
        vm.gridOptions.data = res
        vm.all_metrics = _.chain(res).map(test => test.all_metrics).flatten().uniq().sort().value();
        // Apply options data to filter drop downs
etc...
```

Reply:
Good catch, fixed